src/net/sockopt.rs

```diff
@@ -647,6 +647,50 @@ pub fn ipv6_v6only<Fd: AsFd>(fd: Fd) -> io::Result<bool> {
     backend::net::sockopt::ipv6_v6only(fd.as_fd())
 }
 
+/// `setsockopt(fd, IPPROTO_IP, IP_MULTICAST_IF, value)`
+///
+/// See the [module-level documentation] for more.
+///
+/// [module-level documentation]: self#references-for-get_ip_-and-set_ip_-functions
+#[inline]
+#[doc(alias = "IP_MULTICAST_IF")]
+pub fn set_ip_multicast_ifv4<Fd: AsFd>(fd: Fd, value: &Ipv4Addr) -> io::Result<()> {
+    backend::net::sockopt::set_ip_multicast_ifv4(fd.as_fd(), value)
+}
+
```

Comment:
The [Linux docs](https://man7.org/linux/man-pages/man7/ip.7.html) for `IP_MULTICAST_IF` say "The argument for [setsockopt(2)](https://man7.org/linux/man-pages/man2/setsockopt.2.html) is an ip_mreqn or (since Linux 3.5) ip_mreq structure similar to IP_ADD_MEMBERSHIP, or an
 in_addr structure.". Would it make sense here to also add `set_ip_multicast_ifv4_with_ifindex`, similar to the existing `set_ip_add_membership_with_ifindex`, which uses the `ip_mreqn` form?

Reply:
Yes, add it.